### ISSUE TITLE
feat: Use a GitHub app instead of a bot account for elevated tokens

### ISSUE THREAD

**Issue by erezrokah** [cloudquery/.github#184]
## Context

We use a GitHub personal access token from a bot (`cq-bot`) account when we need to run GitHub actions with elevated permissions.
There are several reasons not to use the built-in GitHub Actions `GITHUB_TOKEN`:
1. Workflows using the built-in `GITHUB_TOKEN` can't trigger new workflow runs. See [here](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow). This means that if we create release PRs, renovate PRs, etc. they won't trigger any tests or other CI workflows (blocking them from getting merged due to branch protection).
2. The built-in `GITHUB_TOKEN` is scoped to the repo it runs in, so we can't use it for cross repo automation (e.g. open PRs in other repos from this repo).
3. When executed from forked repositories, the built-in `GITHUB_TOKEN` has only read permissions. We use the `pull_request_target` event to handle such cases, see https://github.com/cloudquery/.github/blob/62af2e427e7e5f748f87bf849ca197fbc0318891/workflows/providers/test_integration.yml#L10.

## Problem
Using a bot account works well for our use cases, however it has a few downsides:
1. We might hit GitHub rate limits if we have more automation (limit is 5000 requests per hour)
2. Permissions for personal access tokens are not very granular
3. The bot user account takes up a seat in our GitHub organization and we need to save its credentials. See [here](https://docs.github.com/en/developers/apps/getting-started-with-apps/differences-between-github-apps-and-oauth-apps#machine-vs-bot-accounts)
4. Personal access tokens are long living tokens which pose a security risk

TLDR: The main issue I've experienced is with rate limits, as the limits are per account and not per token.

## Suggested solution
We can create a GitHub app and use it for generating tokens. See [here](https://github.com/tibdex/github-app-token/tree/c95b1c441a4dacd6d24f231b9697a5f9a04c607e#example-workflow).

Advantages:
1. Better rate limits for GitHub apps
2. More granular permissions for GitHub apps (can be installed on specific repos too)
3. GitHub app generated tokens expire after 1 hour
4. GitHub apps don't take up a seat

**Comment by erezrokah** [comment]
Moved this to the backlog as not a priority. I opened the issue mostly to document the GitHub app approach as a solution if we ever hit rate limits with our current approach.

**Comment by erezrokah** [comment]
Closing for now. We can re-open if needed
